okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/OkHttpClients.java

```diff
@@ -49,16 +48,7 @@
      */
     private static final ExecutorService executorService = Tracers.wrap(new Dispatcher().executorService());
 
-    /**
-     * The {@link ScheduledExecutorService} used for scheduling call retries. This thread pool is distinct from OkHttp's
-     * internal thread pool and from the thread pool used by {@link #executorService}.
-     * <p>
-     * Note: In contrast to the {@link java.util.concurrent.ThreadPoolExecutor} used by OkHttp's {@link
-     * #executorService}, {@code corePoolSize} must not be zero for a {@link ScheduledThreadPoolExecutor}, see its
-     * Javadoc.
-     */
-    private static final ScheduledExecutorService scheduledExecutorService = Tracers.wrap(
-            new ScheduledThreadPoolExecutor(5, Util.threadFactory("http-remoting/OkHttp Scheduler", false)));
+    private static final ScheduledExecutorService asyncRetryExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
Does this mean all call retries due to `QosIoException` are scheduled on the same thread?